### PR TITLE
adaptively moves offset to the right device so that gpu can be used

### PR DESCRIPTION
**Pull request by alexhagen**

### CODE REVIEW

File: fft_conv_pytorch/fft_conv.py [hunk 1]
```diff
@@ -83,7 +84,7 @@ def fft_conv(
     dilation_ = to_ntuple(dilation, n=n)
 
     # internal dilation offsets
-    offset = torch.zeros(1, 1, *dilation_)
+    offset = torch.zeros(1, 1, *dilation_).to(signal.device)
```
Comment (fkodom) [inline fft_conv_pytorch/fft_conv.py]: For efficiency, can you initialize directly on the device?
```python
offset = torch.zeros(1, 1, *dilation_, device=signal.device, dtype=signal.dtype)
```

^^ The `dtype` argument wasn't being used before, but I'm thinking that's a good idea while we're at it?

**Comment by fkodom** [comment]
Thanks for catching this! Generally, direct convolutions are much faster on GPU unless the kernel is REALLY large, so I hadn't focused much on GPU compatibility. This is a great addition, though, and definitely should be included.

Left you a quick comment -- just a nit-picky detail for initializing directly on the device.

File: fft_conv_pytorch/fft_conv.py (hunk 1 above)
Comment (alexhagen) [reply fft_conv_pytorch/fft_conv.py]: Will do!

**Comment by alexhagen** [comment]
That makes sense - I'm working on using really large kernels, so I guess this is scratching my own itch.

File: fft_conv_pytorch/fft_conv.py (hunk 1 above)
Comment (alexhagen) [reply fft_conv_pytorch/fft_conv.py]: Done.

**Comment by fkodom** [comment]
Excellent! Running unit tests now -- looks like `padding_mode='zeros'` isn't a valid option for some PyTorch versions? I believe `padding_mode='constant'` works correctly, and is backwards compatible. Maybe double-check on your end, and I can merge once we figure that out?

**Review by fkodom — APPROVED** [review]
🎉

**Pull request merged**